pandas/indexes/base.py

```diff
@@ -2630,6 +2630,20 @@ def _join_multi(self, other, how, return_indexers=True):
         raise NotImplementedError("merging with both multi-indexes is not "
                                   "implemented")
 
+    def _array_putmask(self, a, mask, values):
+        """
+        Must be overridden by Indexes which have non-ndarray .values (e.g.
+        CategoricalIndex).
```

Comment:
this is fine, but the point is to follow existing stype and just call this ``putmask``